File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    let smallNum = Math.min(...nums)
    let bigNum = Math.max(...nums)
    for(let i = smallNum; i >= 1; i--){
        if(bigNum / i % 1 == 0 && smallNum / i % 1 == 0) return i;
    }
    return 1;
};

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        nums.sort()
        smallNum = nums[0]
        bigNum = nums[len(nums) -1]
        i = smallNum
        while i >= 1 :
            if bigNum / i % 1 == 0 and smallNum / i % 1 == 0:
                return i;
            i = i -1
        return 1;